user-interface/src/main/java/life/qbic/datamanager/views/general/QbicDialog.java

```diff
@@ -0,0 +1,152 @@
+package life.qbic.datamanager.views.general;
+
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.Key;
+import com.vaadin.flow.component.Shortcuts;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import life.qbic.datamanager.views.notifications.SupportsCloseConfirmation;
+import life.qbic.datamanager.views.notifications.WithCloseListener.CloseEvent;
+
+/**
+ * A {@link com.vaadin.flow.component.dialog.Dialog} with additional functionality.
+ * <p>
+ * This class adds functionality to only close on user confirmation. For this simply call
+ * {@link #requireCloseConfirmation()}.
+ *
+ * @see com.vaadin.flow.component.dialog.Dialog
+ * @since 1.4.0
+ */
+public class QbicDialog extends com.vaadin.flow.component.dialog.Dialog implements
+    SupportsCloseConfirmation<QbicDialog, CloseEvent<QbicDialog>> {
+
+  private final List<ListenerRegistration> dialogCloseListeners = new ArrayList<>();
+  private ListenerRegistration requireCloseConfirmationListener = null;
+  private boolean ignoresModificationForCloseConfirmation = SupportsCloseConfirmation.super.ignoresCloseCheckIfUnmodified();
+  private boolean modified = false;
+
+  public QbicDialog() {
+    setCloseOnOutsideClick(false);
+    setCloseOnEsc(false);
+    Shortcuts.addShortcutListener(this, this::close, Key.ESCAPE);
```

Comment:
this allows closing via ESC for all QbicDialogs, even those not requiring confirmation. Do we want that behaviour?

Reply:
Due to the interface introduces, the close confirmation does not care whether you close by pressing ESC or calling `close`. By adding the shortcut explicitly, the `close` method is always called (triggering close listeners that show the close confirmation).